File: src/main/java/com/teambrmodding/assistedprogression/common/blocks/BlockCrafter.java
package com.teambrmodding.assistedprogression.common.blocks;

import com.teambr.bookshelf.common.IOpensGui;
import com.teambrmodding.assistedprogression.client.gui.GuiCrafter;
import com.teambrmodding.assistedprogression.common.container.ContainerCrafter;
import com.teambrmodding.assistedprogression.common.tiles.TileCrafter;
import net.minecraft.block.material.Material;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;

/**
 * This file was created for AssistedProgression
 * <p>
 * AssistedProgression is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis - pauljoda
 * @since 11/12/17
 */
public class BlockCrafter extends BaseBlock implements IOpensGui {

    /**
     * Constructor
     */
    public BlockCrafter() {
        super(Material.WOOD, "block_crafter", TileCrafter.class);
    }

    /*******************************************************************************************************************
     * IOpensGui                                                                                                       *
     *******************************************************************************************************************/

    /**
     * Return the container for this tile
     *
     * @param id     Id, probably not needed but could be used for multiple guis
     * @param player The player that is opening the gui
     * @param world  The world
     * @param x      X Pos
     * @param y      Y Pos
     * @param z      Z Pos
     * @return The container to open
     */
    @Override
    public Object getServerGuiElement(int id, EntityPlayer player, World world, int x, int y, int z) {
        return new ContainerCrafter(player.inventory, (TileCrafter) world.getTileEntity(new BlockPos(x, y, z)));
    }

    /**
     * Return the gui for this tile
     *
     * @param id     Id, probably not needed but could be used for multiple guis
     * @param player The player that is opening the gui
     * @param world  The world
     * @param x      X Pos
     * @param y      Y Pos
     * @param z      Z Pos
     * @return The gui to open
     */
    @Override
    public Object getClientGuiElement(int id, EntityPlayer player, World world, int x, int y, int z) {
        return new GuiCrafter(player.inventory, (TileCrafter) world.getTileEntity(new BlockPos(x, y, z)));
    }
}

File: src/main/java/com/teambrmodding/assistedprogression/common/blocks/BlockCrafter.scala
package com.teambrmodding.assistedprogression.common.blocks

import com.teambr.bookshelf.common.blocks.traits.DropsItems
import com.teambr.bookshelf.common.tiles.traits.OpensGui
import com.teambrmodding.assistedprogression.AssistedProgression
import com.teambrmodding.assistedprogression.client.gui.GuiCrafter
import com.teambrmodding.assistedprogression.common.container.ContainerCrafter
import com.teambrmodding.assistedprogression.common.tiles.TileCrafter
import net.minecraft.block.material.Material
import net.minecraft.block.state.IBlockState
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.tileentity.TileEntity
import net.minecraft.util.EnumBlockRenderType
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import net.minecraftforge.fml.relauncher.{Side, SideOnly}

/**
 * This file was created for NeoTech
 *
 * NeoTech is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Dyonovan
 * @since August 18, 2015
 */
class BlockCrafter(name: String, tileEntity: Class[_ <: TileEntity]) extends BaseBlock(Material.WOOD, name, tileEntity)
        with OpensGui with DropsItems {

    setHardness(1.5F)
    setCreativeTab(AssistedProgression.tabAssistedProgression)

    override def getRenderType(state : IBlockState) : EnumBlockRenderType = EnumBlockRenderType.MODEL

    override def getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = {
        new ContainerCrafter(player.inventory, world.getTileEntity(new BlockPos(x, y, z)).asInstanceOf[TileCrafter])
    }

    @SideOnly(Side.CLIENT)
    override def getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = {
        new GuiCrafter(player.inventory, world.getTileEntity(new BlockPos(x, y, z)).asInstanceOf[TileCrafter])
    }
}
